main.tf

```
@@ -15,99 +15,123 @@
-  backend         = ["${var.backends["${count.index}"]}"]
-  health_checks   = ["${element(google_compute_http_health_check.default.*.self_link, count.index)}"]
-  security_policy = "${var.security_policy}"
-  enable_cdn      = "${var.cdn}"
+  port_name       = split(",", var.backend_params[count.index])[1]
+  protocol        = var.backend_protocol
+  timeout_sec     = split(",", var.backend_params[count.index])[3]
+  dynamic "backend" {
+    for_each = var.backends[count.index]
+    content {
+      balancing_mode               = lookup(backend.value, "balancing_mode", null)
+      capacity_scaler              = lookup(backend.value, "capacity_scaler", null)
+      description                  = lookup(backend.value, "description", null)
+      group                        = lookup(backend.value, "group", null)
+      max_connections              = lookup(backend.value, "max_connections", null)
+      max_connections_per_instance = lookup(backend.value, "max_connections_per_instance", null)
+      max_rate                     = lookup(backend.value, "max_rate", null)
+      max_rate_per_instance        = lookup(backend.value, "max_rate_per_instance", null)
+      max_utilization              = lookup(backend.value, "max_utilization", null)
+    }
+  }
+  health_checks   = [
+    google_compute_http_health_check.default[count.index].self_link]
+  security_policy = var.security_policy
+  enable_cdn      = var.cdn
 }
 
 resource "google_compute_http_health_check" "default" {
-  project      = "${var.project}"
-  count        = "${length(var.backend_params)}"
+  project      = var.project
+  count        = length(var.backend_params)
   name         = "${var.name}-backend-${count.index}"
-  request_path = "${element(split(",", element(var.backend_params, count.index)), 0)}"
-  port         = "${element(split(",", element(var.backend_params, count.index)), 2)}"
+  request_path = split(",", var.backend_params[count.index])[0]
+  port         = split(",", var.backend_params[count.index])[2]
 }
 
-# Create firewall rule for each backend in each network specified, uses mod behavior of element().
 resource "google_compute_firewall" "default-hc" {
-  count         = "${length(var.firewall_networks) * length(var.backend_params)}"
-  project       = "${element(var.firewall_projects, count.index) == "default" ? var.project : element(var.firewall_projects, count.index)}"
+  count         = length(var.firewall_networks)
+  project       = length(var.firewall_networks) == 1 && var.firewall_projects[0] == "default" ? var.project : var.firewall_projects[count.index]
   name          = "${var.name}-hc-${count.index}"
-  network       = "${element(var.firewall_networks, count.index)}"
-  source_ranges = ["130.211.0.0/22", "35.191.0.0/16", "209.85.152.0/22", "209.85.204.0/22"]
-  target_tags   = ["${var.target_tags}"]
+  network       = length(var.firewall_networks) == 1 ? var.firewall_networks[0] : var.firewall_networks[count.index]
```

This logic is redundant since the count index would be 0 if there is 1 firewall network.

```suggestion
  network       = var.firewall_networks[count.index]
```

---

@aaron-lane thank you fore review! [Fixed](https://github.com/terraform-google-modules/terraform-google-lb-http/compare/acce7ef74273407652471f19ccfbafa12a51248f..42602b96de3e1eaa0f974e3f613a228edf64d322)